DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -72,6 +72,10 @@ internal class DataUploadWorker: DataUploadWorkerType {
                 return
             }
             let context = contextProvider.read()
+            guard !context.isExtension else {
+                // Prevent uploading from app extensions
+                return
+            }
```

Comment:
blocking-til-confirmation/ Is SDK's write & read location shared between app and extension uniformly? What is written in [App Extension Programming Guide](https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/ExtensionScenarios.html) seems to contradict it:

> #### Sharing Data with Your Containing App
> Even though an app extension bundle is nested within its containing app’s bundle, the running **app extension and containing app have no direct access to each other’s containers**.

"Data sharing" can be enabled **optionally** as an extra entitlement granted to the app. We can't assume every app with extension does it:

> You can, however, enable data sharing. For example, you might want to allow your app extension and its containing app to share a single large set of data, such as prerendered assets.
>
> To enable data sharing, use Xcode or the Developer portal to enable app groups for the containing app and its contained app extensions. Next, register the app group in the portal and specify the app group to use in the containing app. To learn about working with app groups, see [Adding an App to an App Group](https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/EntitlementKeyReference/Chapters/EnablingAppSandbox.html#//apple_ref/doc/uid/TP40011195-CH4-SW19).

Looking at [App Groups docs](https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/EntitlementKeyReference/Chapters/EnablingAppSandbox.html#//apple_ref/doc/uid/TP40011195-CH4-SW19), it appears that only the `com.apple.security.application-groups` can enable sharing `caches/` location where SDK stores data:

> The `com.apple.security.application-groups` ... allows multiple apps produced by a single development team to share access to a special group container. This container is intended for content that is not user-facing, such as shared **caches** or databases.

---

There are multiple layers of data sharing to consider in order to validate this change. Once confirmed, we should document this decision in the PR description so we're clear on assumptions we take.